### PR TITLE
v2podman image sign

### PR DESCRIPTION
**Pull request by baude**
this is a straight port to add the podman image sign command. no improvements or refactoring done

Signed-off-by: Brent Baude <bbaude@redhat.com>

### CODE REVIEW

**Comment by TomSweeneyRedHat** [comment]
Vendor issue?  Changes LGTM fwiw.
```
cd ${CIRRUS_WORKING_DIR} && ./hack/tree_status.sh
tree is dirty, please sync the vendor.conf and commit all changes.
```

File: pkg/domain/infra/abi/images.go [hunk 1]
```diff
@@ -549,3 +560,152 @@ func (ir *ImageEngine) Shutdown(_ context.Context) {
...
+		}
+
+		// create the signstore file
+		rtc, err := ir.Libpod.GetConfig()
+		if err != nil {
+			return nil, err
+		}
+		newImage, err := ir.Libpod.ImageRuntime().New(ctx, signimage, rtc.Engine.SignaturePolicyPath, "", os.Stderr, &dockerRegistryOptions, image.SigningOptions{SignBy: options.SignBy}, nil, util.PullImageMissing)
+		if err != nil {
+			return nil, errors.Wrapf(err, "error pulling image %s", signimage)
```
Comment (rhatdan) [inline pkg/domain/infra/abi/images.go:616]: Is it actually pulling the image or is this a bogus wrap?

File: pkg/domain/infra/abi/images.go [hunk 2]
```diff
@@ -549,3 +560,152 @@ func (ir *ImageEngine) Shutdown(_ context.Context) {
...
+		}
+
+		if rootless.IsRootless() {
```
Comment (rhatdan) [inline pkg/domain/infra/abi/images.go:619]: This code should be greatly simplified:

if sigStoreDir == "" {
                 if rootless.IsRootless() {
				sigStoreDir = filepath.Join(filepath.Dir(ir.Libpod.StorageConfig().GraphRoot), "sigstore") 
               } else {
			registryInfo := trust.HaveMatchRegistry(rawSource.Reference().DockerReference().String(), registryConfigs)
￼			if registryInfo != nil {
                                        if sigStoreDir = registryInfo.SigStoreStaging; sigStoreDir == "" {
￼						sigStoreDir = registryInfo.SigStore
￼					}
              }
} 

sigStoreDir, err = isValidSigStoreDir(sigStoreDir)
￼if err != nil {
		return nil, errors.Wrapf(err, "invalid signature storage %s", sigStoreDir)
}

File: pkg/domain/infra/abi/images.go (hunk 1 above)
Comment (baude) [reply pkg/domain/infra/abi/images.go:616]: look to me like it will pull it

File: pkg/domain/infra/abi/images.go (hunk 2 above)
Comment (baude) [reply pkg/domain/infra/abi/images.go:619]: want me to change it? like i said in the commit, i did a straight port. no refactoring

Comment (rhatdan) [reply pkg/domain/infra/abi/images.go:619]: Yes, I think you could easily simplify this code.

**Comment by baude** [comment]
needs merge love

**Pull request merged**